clippy_lints/src/utils/higher.rs

```diff
@@ -114,13 +114,36 @@ pub fn range(expr: &hir::Expr) -> Option<Range> {
 
 /// Checks if a `let` decl is from a `for` loop desugaring.
 pub fn is_from_for_desugar(decl: &hir::Decl) -> bool {
+    // This will detect plain for-loops without an actual variable binding:
+    //
+    // ```
+    // for x in some_vec {
+    //   // do stuff
+    // }
+    // ```
     if_let_chain! {[
         let hir::DeclLocal(ref loc) = decl.node,
         let Some(ref expr) = loc.init,
         let hir::ExprMatch(_, _, hir::MatchSource::ForLoopDesugar) = expr.node,
     ], {
         return true;
     }}
+
+    // This detects a variable binding in for loop to avoid `let_unit_value`
```

Comment:
For the previous comment I had the println!'s wrong. There are a couple of tests but the example `for Foo { a } in bar` is handled by the original check in `is_from_for_desugar`.